### DISCUSSION TITLE
How to add a store as a Singleton?  in Blazor Server

### DISCUSSION THREAD

**Discussion by JimiSweden** [mrpmorris/Fluxor#367] [Q&A]
Hi Peter (et al.)

How can I add a store as a Singleton? 
I tried to add the option - WithLifetime(StoreLifetime.Singleton);
However this does not work and throws the error ...

```
Fluxor.Blazor.Web.Middlewares.Routing.RoutingMiddleware': Cannot consume scoped service 'Microsoft.AspNetCore.Components.NavigationManager' from singleton 
```

The Microsoft.AspNetCore.Components.NavigationManager 
is registered as a Singleton in Blazor WASM , but as  Scoped in Blazor Server 
https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/dependency-injection?view=aspnetcore-7.0#default-services

Thanks!
Jimi

**Comment by mrpmorris** [comment]
I doubt you can use Routing middleware if you register as a singleton.

**Reply by JimiSweden** [reply]
Thanks, I realized it doesnt work with reduxdevtools either.

Perhaps I'm doing it wrong. 
I want to have a 'globally' shared state with, in it's simplest form,  a list of items.
And the web clients should get updated from, and push updates to, the same state. 

But perhaps I need to use other event subsriptions for that (in a shared singleton "Global state")

Do you have any similar implementations or recommendations? 
(I'm fairly new to Blazor, but have some experience from Angular and React)

Thanks!

**Reply by mrpmorris** [reply]
If it's a WASM app then scoped dependencies are reused throughout your UI session. The store is registered as Scoped by default.

So they are effectively singletons.
